File: android-app/feature/feature-player/src/main/java/tv/bayit/plus/feature/player/dialogue/PauseAskModels.kt
package tv.bayit.plus.feature.player.dialogue

/**
 * Phases of the Pause-to-Ask dialogue flow.
 *
 * Mirrors the iOS `PauseAskPhase` enum for feature parity.
 * The user progresses through these phases sequentially:
 * character selection, question input, server processing,
 * avatar lip-sync playback, transition, character response, and idle.
 */
enum class PauseAskPhase {
    SELECTING,
    INPUT,
    POLISHING,
    USER_SPEAKING,
    TRANSITION,
    CHARACTER_SPEAKING,
    IDLE,
}

File: ios-app/BayitPlusApp/Models/PauseAskModels.swift
import Foundation

/// Response from the Pause & Ask backend endpoint.
/// Contains animated lip-sync videos for both the user avatar and character.
struct PauseAskResponse: Codable {
    let userPolishedText: String
    let userAudioUrl: String
    let userAnimatedVideoUrl: String
    let userVideoDuration: Double
    let characterName: String
    let characterResponseText: String
    let characterAudioUrl: String
    let characterAnimatedVideoUrl: String
    let characterVideoDuration: Double
}

/// Tracks the current phase of a Pause & Ask interaction.
enum PauseAskPhase: String {
    case selecting
    case input
    case polishing
    case userSpeaking
    case transition
    case characterSpeaking
    case idle
    case error
}
